lib/collection/variable.js

```diff
@@ -115,26 +116,64 @@ _.assign(Variable.prototype, /** @lends Variable.prototype */ {
      * @returns {*}
      */
     cast: function (value) {
-        return (Variable.types[this.type] || Variable.types.any)(value);
+        return this.castOut(value);
+    },
+
+    /**
+     * Typecasts a value to the {@link Variable.types} of this {@link Variable}. Returns the value of the variable
+     * converted to the type specified in {@link Variable#type}.
+     *
+     * @private
+     * @param {*} value
+     * @returns {*}
+     */
+    castIn: function (value) {
```

Comment:
Extremely similar to `castOut`, could be wrapped around a generic `cast` function.

Reply:
I’m afraid that wouldn’t be possible. There’s already a cast function which is public and that is exactly same as .castOut

Also - this is very low level function. It shouldn’t hurt and may even give more performance.